Orbit.h

```diff
@@ -36,6 +36,8 @@
 #define ORBIT_START_ASYNC(name, id) ORBIT_START_ASYNC_WITH_COLOR(name, id, orbit::Color::kAuto)
```

Comment:
Sorry for asking some basic questions, I haven't been following this closely. 

Do "name" here and "str" play the same role? If yes, I'm not sure how, as a user, I would combine ORBIT_ASYNC_STRING and ORBIT_START_ASYNC. I assume that ORBIT_ASYNC_STRING just overrides the string that I pass as "name"?

Reply:
Looking at the code below, I guess as a  user, when I have registered a string with ORBIT_ASYNC_STRING, I'll just pass nullptr as "name", correct?

Is the idea to also support arbitrary length strings for the non-async case in the same way?

Reply:
"name" here is actually the track name, I should rename the variable to make it clear. A user would never pass nullptr, no. ASYNC_STRING is used as a description for individual timeslices. For example, let's say we wanted to track file IO operations, the "name" (name of the track) would be "File IO operations" and the ASYNC_STRING macro would be used to pass in the full name of the requested file. All file IO operations will be under one single track, but we'll have information about the individual operations.

As for arbitrarily long strings for non async, since there is significant overhead, I'd rather not support it for now. This will be supported with the upcoming userspace dynamic instrumentation however.

Reply:
I have a PR coming up with proper documentation inside of Orbit.h that should make this clearer.

Reply:
Ok, thanks for all the details!